From 075761520c2742b3395513d5e9fa59935ca2581d Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Wed, 28 Aug 2019 12:21:07 +0530
Subject: [PATCH 09/11] coresight: tmc: Add Marvell OcteonTx CN9xxx TMC device

commit 396e26d0a6b7b5355647b9f28b94fba51fd32f22 from
git@git.assembla.com:cavium/WindRiver.linux.git

Add device id for Marvell OcteonTx CN9xxx TMC device.

Change-Id: I0ce1a0daf25452bb1791b68da02a8405ada47587
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/16635
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/hwtracing/coresight/coresight-tmc.c | 6 ++++++
 drivers/hwtracing/coresight/coresight-tmc.h | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/hwtracing/coresight/coresight-tmc.c b/drivers/hwtracing/coresight/coresight-tmc.c
index 456f122df74f..cddd41b82fed 100644
--- a/drivers/hwtracing/coresight/coresight-tmc.c
+++ b/drivers/hwtracing/coresight/coresight-tmc.c
@@ -450,6 +450,12 @@ static const struct amba_id tmc_ids[] = {
 		.id	= 0x000bb9ea,
 		.mask	= 0x000fffff,
 	},
+	{
+		/* Marvell OcteonTx CN9xxx */
+		.id	= OCTEONTX_CN9XXX_ETR,
+		.mask	= 0x000fffff,
+		.data	= (void *)(unsigned long)OCTEONTX_CN9XXX_ETR_CAPS,
+	},
 	{ 0, 0},
 };
 
diff --git a/drivers/hwtracing/coresight/coresight-tmc.h b/drivers/hwtracing/coresight/coresight-tmc.h
index dfaff077a7fc..86739d0ac15a 100644
--- a/drivers/hwtracing/coresight/coresight-tmc.h
+++ b/drivers/hwtracing/coresight/coresight-tmc.h
@@ -122,6 +122,12 @@ enum tmc_mem_intf_width {
 #define CORESIGHT_SOC_600_ETR_CAPS	\
 	(TMC_ETR_SAVE_RESTORE | TMC_ETR_AXI_ARCACHE)
 
+/* Marvell OcteonTx CN9xxx TMC-ETR unadvertised capabilities */
+#define OCTEONTX_CN9XXX_ETR_CAPS	\
+	(TMC_ETR_SAVE_RESTORE)
+
+/* Marvell OcteonTx CN9xxx device */
+#define OCTEONTX_CN9XXX_ETR		0x000cc213
 /**
  * struct tmc_drvdata - specifics associated to an TMC component
  * @base:	memory mapped base address for this component.
-- 
2.17.1

